slack/dialog_handlers.py

```diff
@@ -18,12 +18,20 @@ def report_incident(user_id: str, channel_id: str, submission: json, response_ur
     report = submission['report']
     summary = submission['summary']
     impact = submission['impact']
-    lead = submission['lead']
+    lead_id = submission['lead']
     severity = submission['severity']
+    lead = None
+
+    name = get_user_profile(user_id)['name']
+    reporter, created = UserExternalID.objects.get_or_create(app_id='slack', external_id=user_id, display_name=name)
```

Comment:
We repeat this line a lot in the Slack app.  Could we have a helper function somewhere to create a slack user?  
```suggestion
    reporter = GetOrCreateSlackUser(user_id, display_name=name)
```